### PR TITLE
importinto: remove separate wait routine in detached mode and refactor

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #49008

Problem Summary:

### What changed and how does it work?
- remove separate wait routine in detached mode, move FlushTableStats callsite before FinishJob
- remove some legacy code

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/importinto/job.go [hunk 1]
```diff
@@ -38,105 +38,26 @@ import (
...
-		stmt:           stmt,
-		logger:         logutil.BgLogger(),
-		sourceFileSize: sourceFileSize,
-	}, nil
-}
-
-// NewDistImporterCurrNode creates a new DistImporter to import data on current node.
-func NewDistImporterCurrNode(param *importer.JobImportParam, plan *importer.Plan, stmt string, sourceFileSize int64) (*DistImporter, error) {
+// SubmitStandaloneTask submits a task to the distribute framework that only runs on the current node.
+// when import from server-disk, pass engines too, as scheduler might run on another
```
Comment (lance6716) [inline pkg/disttask/importinto/job.go:42]:
```suggestion
// when import from server-disk, pass engine checkpoints too, as scheduler might run on another
```

File: pkg/executor/import_into.go [hunk 2]
```diff
@@ -209,44 +166,40 @@ func (e *ImportIntoExec) fillJobInfo(ctx context.Context, jobID int64, req *chun
...
-		return importinto.NewDistImporterServerFile(param, e.importPlan, e.stmt, ecp, e.controller.TotalFileSize)
+		return importinto.SubmitStandaloneTask(ctx, e.controller.Plan, e.stmt, ecp)
 	}
 	// if tidb_enable_dist_task=true, we import distributively, otherwise we import on current node.
 	if variable.EnableDistTask.Load() {
-		return importinto.NewDistImporter(param, e.importPlan, e.stmt, e.controller.TotalFileSize)
+		return importinto.SubmitTask(ctx, e.controller.Plan, e.stmt)
 	}
-	return importinto.NewDistImporterCurrNode(param, e.importPlan, e.stmt, e.controller.TotalFileSize)
+	return importinto.SubmitStandaloneTask(ctx, e.controller.Plan, e.stmt, nil)
```
Comment (lance6716) [inline pkg/executor/import_into.go:188]: In which case we will reach this branch? please add comments

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: pkg/executor/import_into.go (hunk 2 above)
Comment (D3Hunter) [reply pkg/executor/import_into.go:188]: i think it's quite clear from the code that it's when disttask is not enabled

**Review by lance6716 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**